src/frontend/src/optimizer/plan_node/logical_scan.rs

```diff
@@ -216,6 +218,34 @@ impl LogicalScan {
         &self.predicate
     }
 
+    /// Return indices of fields the output is ordered by and
+    /// corresponding direction
+    pub fn get_out_column_index_order(&self) -> Vec<FieldOrder> {
+        return self
+            .table_desc
+            .order_key
+            .iter()
+            .map(|order| {
+                let out_idx: Vec<usize> = self
+                    .output_col_idx
+                    .iter()
+                    .enumerate()
+                    .filter(|(_, in_idx)| *in_idx == &order.column_idx)
+                    .map(|(out_idx, _)| out_idx)
+                    .collect();
+                assert!(out_idx.len() < 2);
```

Comment:
although it is a little strange, we do not ban the same column appearing in output_col_idx for multiple times. I think we can give any of the columns index in the output schema in that case.